core/modules/syncer.js

```diff
@@ -138,7 +136,35 @@ Syncer.prototype.createTiddlerInfo = function(title) {
 Checks whether the wiki is dirty (ie the window shouldn't be closed)
 */
 Syncer.prototype.isDirty = function() {
-	return (this.numTasksInQueue() > 0) || (this.numTasksInProgress() > 0);
+	// Check tiddlers that are in the store and included in the filter function
+	var titles = this.filterFn.call(this.wiki);
+	for(var index=0; index<titles.length; index++) {
```

Comment:
Is this efficient? `.isDirty` has been a really simple function. It is now replaced with 2 loops, that both iterate over the complete wiki store.

Reply:
That is indeed a concern, and needs investigation. I'm hoping it may not be a problem: these loops are fast, pure JS and don't hit the DOM. The old implementation computed the same state, just at different times, and went to a lot more trouble to maintain that state, with relatively expensive hashmap insertions and deletions.

Reply:
The situation in https://github.com/Jermolene/TiddlyWiki5/pull/4397 and https://github.com/Jermolene/TiddlyWiki5/issues/4275 involved an Array.indexOf call while iterating through the array, which resulted in length squared indexOf calls. That only became a problem once someone created 60,000 tiddlers, which resulted in 3.6 billion indexOf calls.